### PR TITLE
Improve yaml specification for variance and standard deviation error types

### LINKED ISSUE #16: Read in the standard deviation rather than variance from file
## Description

Inside jopa the background error variable is actually the standard deviation. We would like to update the interface to orca-jedi to make this clearer.

## Acceptance Criteria (Definition of Done)

Updates to the tests run in all systems.

## Dependencies
- sith suite configuration files will need to be updated
- nemo-feedback configuration files will need to be updated
- Other Ocean comparison/extraction suites will need updating

### PR DESCRIPTION
**Pull request by twsearle**
## Description

At the moment both variance and standard deviation are referred to as "error" within UFO. We have been referring to all our errors as variances, when in fact, most of them are standard deviations with only some variances in the case of profile data types.

This has always been very confusing. This change is an attempt to contain this confusion to some extent within the orca-jedi model interface, so that the "variable type" parameter contains the correct information.

## Issue(s) addressed

Resolves #16

## Dependencies

- [x] jjdocs yaml and documentation change MetOffice/jjdocs/pull/204

## Impact

None expected.

## Checklist

- [x] I have updated the unit tests to cover the change
- [x] New functions are documented briefly via Doxygen comments in the code
- [x] I have linted my code using cpplint
- [x] I have run the unit tests
- [x] I have run mo-bundle to check integration with the rest of JEDI and run the unit tests under all environments

### CODE REVIEW

**Review by s-good — CHANGES_REQUESTED** [review]
Thanks for making this change. It will be great to get this sorted.

Please could I confirm how this now works? Does specifying background variance or background standard deviation as the variable type do exactly the same thing? 

I noticed that e.g. in example/hofx3d_odb_prof_example.yaml the variable type is background standard deviation but the nemo field name is t_tot_var which is confusing - not sure if there is something that I'm missing about what it is doing?

Variable type should really be background error variance or background error standard deviation rather than without 'error', as background variance is a different quantity to background error variance.

**Comment by twsearle** [comment]
> Please could I confirm how this now works? Does specifying background variance or background standard deviation as the variable type do exactly the same thing?

yes, for now they are treated the same, I am just trying to make the configurations easier to read for now. In the future we could treat them differently when reading them in for example.

> I noticed that e.g. in example/hofx3d_odb_prof_example.yaml the variable type is background standard deviation but the nemo field name is t_tot_var which is confusing - not sure if there is something that I'm missing about what it is doing?

An error on my part! I will fix it.

> Variable type should really be background error variance or background error standard deviation rather than without 'error', as background variance is a different quantity to background error variance.

OK, I will make that update along with the other change before returning this for review.  That does mean the change will be a coordinated change, as we loose compatibility with the previous name. I think perhaps I don't know what the background error variance is on reflection! Is it the variance in the difference between the truth and the background? Whereas background variance would be quantifying the variablity in the background itself?

**Comment by s-good** [comment]
> I think perhaps I don't know what the background error variance is on reflection! Is it the variance in the difference between the truth and the background? Whereas background variance would be quantifying the variablity in the background itself?

That's basically it. The background error variance tells us about how background errors (the differences between the background and the (unknown) truth) are distributed. The background variance tells us how the background values are distributed.

File: examples/hofx_nc_ice_mpi.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ geometry:
     - name: ice_area_fraction_background_error
       nemo field name: seaice_tot_var
       model space: surface
-      variable type: background standard deviation
+      variable type: background error standard deviation
```
Comment (s-good) [inline examples/hofx_nc_ice_mpi.yaml:18]: Should this be variance to match the nemo field name?

File: src/tests/orca-jedi/test_geometry.cc [hunk 2]
```diff
@@ -32,7 +32,7 @@ CASE("test basic geometry") {
   nemo_var_mappings[1].set("name", "sea_ice_area_fraction_error")
     .set("nemo field name", "sic_tot_var")
     .set("model space", "surface")
-    .set("variable type", "background standard deviation");
+    .set("variable type", "background error standard deviation");
```
Comment (s-good) [inline src/tests/orca-jedi/test_geometry.cc:35]: Should this be variance to match the nemo field name?

File: src/tests/orca-jedi/test_interpolator.cc [hunk 3]
```diff
@@ -65,7 +65,7 @@ CASE("test  interpolator") {
     nemo_var_mappings[1].set("name", "sea_ice_area_fraction_error")
       .set("nemo field name", "sic_tot_var")
       .set("model space", "surface")
-      .set("variable type", "background standard deviation");
+      .set("variable type", "background error standard deviation");
```
Comment (s-good) [inline src/tests/orca-jedi/test_interpolator.cc:68]: Should this be variance to match the nemo field name?

File: src/tests/testinput/hofx3d_nc_potm.yaml [hunk 4]
```diff
@@ -11,7 +11,7 @@ geometry :
     - name: sea_water_potential_temperature_background_error
       nemo field name: t_tot_var
       model space: volume
-      variable type: background standard deviation
+      variable type: background error standard deviation
```
Comment (s-good) [inline src/tests/testinput/hofx3d_nc_potm.yaml:14]: Should this be variance to match the nemo field name?

File: src/tests/testinput/hofx3d_nc_prof_2vars.yaml [hunk 5]
```diff
@@ -11,14 +11,14 @@ geometry :
...
-      variable type: background variance
+      variable type: background error variance
     - name: salinity
       nemo field name: vosaline
       model space: volume
     - name: salinity_background_error
       nemo field name: t_tot_var
       model space: volume
-      variable type: background standard deviation
+      variable type: background error standard deviation
```
Comment (s-good) [inline src/tests/testinput/hofx3d_nc_prof_2vars.yaml:21]: Should this be variance to match the nemo field name?

File: src/tests/testinput/hofx3d_nc_sst.yaml [hunk 6]
```diff
@@ -11,11 +11,11 @@ geometry :
     - name: ice_area_fraction_background_error
       nemo field name: sic_tot_var
       model space: surface
-      variable type: background variance
+      variable type: background error variance
     - name: sea_surface_temperature_background_error
       nemo field name: t_tot_var
       model space: surface
-      variable type: background standard deviation
+      variable type: background error standard deviation
```
Comment (s-good) [inline src/tests/testinput/hofx3d_nc_sst.yaml:18]: Should this be variance to match the nemo field name?

**Review by s-good — CHANGES_REQUESTED** [review]

File: src/tests/orca-jedi/test_geometry.cc (hunk 2 above)
Comment (twsearle) [reply src/tests/orca-jedi/test_geometry.cc]: Sorry about these oversights! I have made the changes in the latest commit. Hopefully that covers all of them.

**Review by s-good — APPROVED** [review]
Thanks for making the changes.

**Review by matthewrmshin — APPROVED** [review]
Tested in my environment. Works as expected.

**Pull request merged**